Lesson15/Busra/busra.js

```diff
@@ -0,0 +1,508 @@
+
+
+  if (normalizeStr === ters) {
+    console.log(str +  "  is a palindrome");
+  } else {
+    console.log(str + "  is not a palindrome");
+  }
+}
+
+isPalindrome("busra");
+isPalindrome("ey Nihat Adana'da tahin ye");  //Adana ve Nihat isimlerini büyük harfle yazdığımızda, Adana'da da kesme işareti kullandığımızda doğru sonucu vermiyor bu yüzden en başta bir düzenleme yapmamız gerek
+
+
+
+
+
+/*
+10. Check Even or Odd (if-else)
+   - Define a function `evenOrOdd(number)` that:
+     - Logs "Even" if the number is even
+     - Logs "Odd" if the number is odd
+*/
+
+function evenOrOdd(number){
+
+  if(number%2===0){
+    console.log("number is even");
+  }
+  else{
+    console.log("number is odd");
+  }
+}
+
+evenOrOdd(224);
+evenOrOdd(7485);
+
+
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+
+function checkProtocol(url){
+  let normalizeUrl = url.toLowerCase();
+  if(normalizeUrl.startsWith("https")){
+    console.log("Secure connection");
+  }
+  else{
+    console.log("Unsecure connection");
+  }
+
+}
+checkProtocol("https://www.freecodecamp.org/");
+
+checkProtocol("hptts://www.freecodecamp.org/");
+
+
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+
+function repeatWord(word, times){
+ 
+  console.log(word.repeat(times));
+}
+repeatWord("busra",15);
+
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+function censorWord(sentence, target){
+  console.log(sentence.replace(target,"****"));
+}
+censorWord("I need to study","need");
+
+
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+
+function startsWithA(str){
+  if(str.charAt(0)==='A'){
+    console.log("Starts with A");
+  }
+  else{
+    console.log("Does not start with A");
+  }
+}
+startsWithA("I need to study");
+
+startsWithA("And I am going to bring my daugther too");
+
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+function sliceLastN(text, n){
+  console.log(text.slice(-n));
+}
+sliceLastN("I have to study",3);
+
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+
+function gradeChecker(score){
+ if(score>=90){
```

Comment:
Well done! It's a good example of situation where if-else statement produces cleaner code than a switch statement. 👍